rebench/executor.py

```diff
@@ -106,14 +106,19 @@ def execute(self):
 
 
 class BatchScheduler(RunScheduler):
-
     def _process_remaining_runs(self, runs):
-        for run_id in runs:
+        remaining_runs = list(runs)
+        while len(remaining_runs) > 0:
+            run_id = remaining_runs[0]
+            remaining_runs.pop(0)
             try:
                 completed = False
                 while not completed:
                     completed = self._executor.execute_run(run_id)
                     self._indicate_progress(completed, run_id)
+                    if run_id.benchmark.suite.executor.binary_missing:
```

Comment:
Could you implement a method on the run_id so that we can change this to the following?

```
if run_id.is_binary_missing():
```

This arguably reads easier.
For a bit more context, and also possible drawbacks see https://en.wikipedia.org/wiki/Law_of_Demeter

So, I'd just have a method in run_id that does `return self.benchmark.suite.executor.binary_missing`.

My reasoning is that the executor should only directly interact with the run_id, not any of the objects it is referring to (the current code may violate this principle already, but I think it would be nice to have it).

The run_id can know more freely about the other objects and their structure since they are already tightly bound together.